src/runner.js

```diff
@@ -13,19 +13,45 @@ class Runner {
     core.info(
       `Commits: Comparing the PR commits to verify that they are all from ${privileged_requester_username}`,
     );
-    for (const [, commit] of Object.entries(this.pullRequest.listCommits())) {
-      let commitAuthor = commit.author.login.toLowerCase();
+
+    const useCommitVerification = core.getBooleanInput("commitVerification");
+    const commits = this.pullRequest.listCommits();
+    let allCommitsVerified = true;
+
+    for (const commit of commits) {
+      const commitAuthor = commit.author.login.toLowerCase();
+      const commitVerification = commit?.verification?.verified;
+
+      // check if the commit is verified
+      if (!commitVerification) {
+        allCommitsVerified = false;
+        if (useCommitVerification === true) {
+          core.warning("Unexpected unverified commit");
```

Comment:
Could you add the commit hash here?
```suggestion
          core.warning(`Unexpected unverified commit ${commit.hash}`);
```

Reply:
I think that would be a good idea. Although I think it is [`commit.sha`](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request) instead of `commit.hash`